components/style/gecko/wrapper.rs

```diff
@@ -2066,6 +2065,15 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
                     )
                 }
             }
+            NonTSPseudoClass::Dir(ref dir) => {
+                unsafe {
+                    Gecko_MatchStringArgPseudo(
```

Comment:
The point of converting to an `enum` is not having to go through `Gecko_MatchStringArgPseudo`.

Instead of that, we can just do:

```rust
match *dir {
   Direction::Ltr => self.get_state().intersects(element_state::IN_HAS_DIR_ATTR_LTR_STATE),
   Direction::Rtl => self.get_state().intersects(element_state::IN_HAS_DIR_ATTR_RTL_STATE),
   Direction::Other(..) => false,
}
```